### PR TITLE
Add initial registry tests

### PR DESCRIPTION
**Pull request by denis-tingajkin**
Signed-off-by: Denis Tingajkin <denis.tingajkin@xored.com>

## Motivation

https://github.com/networkservicemesh/integration-k8s-kind/issues/7

### CODE REVIEW

**Comment by denis-tingajkin** [comment]
Found a few bugs:

1. https://github.com/networkservicemesh/integration-k8s-kind/issues/7
2. grpcutils.ListenAndServer should return zero size channel in case if server has started
3. Need to add some utils method for waiting socket ready to connect as we have in monorepo.
4. https://github.com/networkservicemesh/sdk/pull/441

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -140,6 +140,10 @@ jobs:
           gotestsum_version: 0.4.1
       - name: Check kind cluster
         run: |
+          docker build cmd/nse -t networkservicemeshci/nse:master
+          docker build cmd/fake-nsmgr -t networkservicemeshci/fake-nsmgr:master
```
Comment (edwarnicke) [inline .github/workflows/ci.yaml:144]: Why are we using a fake-nsmgr instead of the real one?

Comment (denis-tingajkin) [reply .github/workflows/ci.yaml:144]: @edwarnicke we'll replace this fake-nsmgr with real. As I know @haiodo already adding real nsmgr tests.

File: basic_suite_test.go [hunk 2]
```diff
@@ -48,23 +53,85 @@ func (s *BasicTestsSuite) TearDownSuite() {
...
+
+	s.Require().NoError(exechelper.Run("kubectl apply -f ./deployments/registry-memory.yaml", s.options...))
+	s.Require().NoError(exechelper.Run("kubectl wait --timeout=120s  --for=condition=ready pod -l app=nsm-registry", s.options...))
+
+	s.Require().NoError(exechelper.Run("kubectl apply -f ./deployments/nse.yaml", s.options...))
+	s.Require().NoError(exechelper.Run("kubectl wait --timeout=120s  --for=condition=ready pod -l app=nse", s.options...))
+
+	s.Require().NoError(exechelper.Run("kubectl wait --timeout=120s  --for=condition=ready pod -l app=nsm-registry-proxy-dns", s.options...))
+
+	s.Require().NoError(k8s.ApplyDeployment("./deployments/fake-nsmgr.yaml", func(nsmgr *v1.Deployment) {
```
Comment (edwarnicke) [inline basic_suite_test.go:110]: Curious why the k8s api here for deployment rather than kubectl ?

Comment (denis-tingajkin) [reply basic_suite_test.go:110]: As I know with yaml models we can't just via kubectl dynamically edit properties.

Comment (denis-tingajkin) [reply basic_suite_test.go:110]: Probably we can prepare yaml models per tests as separate files

Comment (denis-tingajkin) [reply basic_suite_test.go:110]: As an alternative we can switch to using helm, then we'll able to use setvariavles in cmd 
@edwarnicke Thoughts?

**Comment by edwarnicke** [comment]
> 2\. grpcutils.ListenAndServer should return zero size channel in case if server has started

I'm curious why this should be so?

**Comment by denis-tingajkin** [comment]
> > 2. grpcutils.ListenAndServer should return zero size channel in case if server has started
> 
> I'm curious why this should be so?

Oops! this issue is not actual more :)

File: basic_suite_test.go (hunk 2 above)
Comment (edwarnicke) [reply basic_suite_test.go:110]: I guess the question is: why do we need to edit this yaml file?  Why isn't it good to go out of the box?

File: cmd/nse/main.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
```
Comment (edwarnicke) [inline cmd/nse/main.go:1]: Why are we building cmd/ s into the integration tests rather than using cmd-* repos?

**Pull request merged**